src/nspawn/nspawn-register.c

```diff
@@ -132,7 +132,11 @@ int register_machine(
                                            * container to ever create
                                            * these device nodes. */
                                           "/dev/pts/ptmx", "rw",
-                                          "char-pts", "rw");
+                                          "char-pts", "rw",
+                                          /* Allow /run/systemd/inaccessible/{chr,blk}
+                                           * devices inside the container */
+                                          "/run/systemd/inaccessible/chr", "rwm",
+                                          "/run/systemd/inaccessible/blk", "rwm");
```

Comment:
I thought the same thing, but I do not like programs that write on on the
directories program

On Thursday, July 21, 2016, Lennart Poettering <notifications@github.com>
wrote:

> In src/nspawn/nspawn-register.c
> <https://github.com/systemd/systemd/pull/3773#discussion_r71740062>:
>
> > @@ -132,7 +132,11 @@ int register_machine(
> >                                             * container to ever create
> >                                             * these device nodes. */
> >                                            "/dev/pts/ptmx", "rw",
> > -                                          "char-pts", "rw");
> > +                                          "char-pts", "rw",
> > +                                          /* Allow /run/systemd/inaccessible/{chr,blk}
> > +                                           * devices inside the container */
> > +                                          "/run/systemd/inaccessible/chr", "rwm",
> > +                                          "/run/systemd/inaccessible/blk", "rwm");
>
> an easy hack in rkt could be to simply create the
> /run/systemd/inaccessible/{blk|chr} device nodes on its own, if it sees
> they are missing, before invoking nspawn...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/systemd/systemd/pull/3773/files/c61816d067ce8f7fd816b33613e9e2ad01b7b76b#r71740062>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGRc70ip8uICF-qwCtT0Ph4g-qKx-5Osks5qX593gaJpZM4JRx03>
> .
>


-- 

Alessandro Puccetti